### PR TITLE
fix(ButtonWithOptions): doesn't invoke onSelect when same option is s…

### PR DESCRIPTION
**Pull request by matveyok**
when no selectedId presented on ButtonWithOptions onSelect invokes on same option.

### CODE REVIEW

**Comment by shlomitc** [comment]
Looks like the core issue is in the `DropdownLayout` component and should be handled there.
https://github.com/wix/wix-style-react/blob/master/src/DropdownLayout/DropdownLayout.js#L53

The `selectedId` prop should be handled as an initial value.

In the case above, we compare the newly selected id with the current prop (the initial value).
The right implementation will be comparing it with the current state:
```
options[index] && onSelect && onSelect(options[index], options[index].id === this.state.selectedId);
```

I pushed my change (will also see that it passes on CI).
Any thoughts about it?

**Comment by matveyok** [comment]
@argshook can you check this please?

**Comment by matveyok** [comment]
@argshook can you check this please?

**Comment by matveyok** [comment]
@shlomitc can you merge this :) ?

File: src/DropdownLayout/DropdownLayout.js [hunk 1]
```diff
@@ -45,12 +45,12 @@ class DropdownLayout extends WixComponent {
...
 
   _onSelect(index) {
-    const {options, onSelect, selectedId} = this.props;
+    const {options, onSelect} = this.props;
     this.setState({
       selectedId: options[index] ? options[index].id : undefined,
       hovered: NOT_HOVERED_INDEX
     });
-    options[index] && onSelect && onSelect(options[index], options[index].id === selectedId);
+    options[index] && onSelect && onSelect(options[index], options[index].id === this.state.selectedId);
```
Comment (argshook) [inline src/DropdownLayout/DropdownLayout.js]: it is possible that by the time line 53 is evaluated, the value of `this.state.selectedId` could be wrong.

probably should be using `this.setState({}, () => /* this callback */)`

Comment (shlomitc) [reply src/DropdownLayout/DropdownLayout.js]: Actually, it should be the other way around. it should check for the `selectedId` state value before the state change

**Comment by shlomitc** [comment]
@argshook - I did a small refactor for code and test, please review

**Pull request merged**